libvirt/domain.go

```diff
@@ -737,6 +750,57 @@ func setNetworkInterfaces(d *schema.ResourceData, domainDef *libvirtxml.Domain,
 	return nil
 }
 
+func tagsToXML(tags map[string]interface{}, metadata *TerraformInstanceXML) (out string, err error) {
+	// Overwrite existing tags while keeping additional metadata
+	metadata.Tags = []TerraformTagXML{}
+	for key, value := range tags {
+		metadata.Tags = append(metadata.Tags, TerraformTagXML{
+			Key:   key,
+			Value: value.(string),
+		})
+	}
+	var bytesOut []byte
+	if bytesOut, err = xml.Marshal(metadata); err != nil {
+		return "", fmt.Errorf("Failed to marshal metadata XML: %s", err)
+	}
+	return string(bytesOut), nil
+}
+
+func setMetadata(d *schema.ResourceData, domain *libvirtxml.Domain) (err error) {
+	metadata := TerraformInstanceXML{}
+	if err = xml.Unmarshal([]byte(domain.Metadata.XML), &metadata); domain.Metadata.XML != "" && err != nil {
+		return fmt.Errorf("XML Unmarshal Error: %s", err)
+	}
+	if _, ok := d.GetOk("tags"); ok {
+		tags := d.Get("tags").(map[string]interface{})
+		var out string
```

Comment:
this afaik you can omit it by using `:=` next line

Reply:
I use `out` after the error check. with `:=` it would only available inside the if{} context.